src/main/java/ru/testtask/service/FileService.java

```diff
@@ -0,0 +1,250 @@
+import ru.testtask.dto.UploadFileDTO;
+import ru.testtask.exception.FileIsToLargeException;
+import ru.testtask.exception.NameAlreadyExistsException;
+import ru.testtask.model.FileData;
+import ru.testtask.repo.FileDataRepo;
+
+import javax.annotation.PostConstruct;
+import javax.servlet.http.HttpServletResponse;
+import java.io.*;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+public class FileService {
+
+    @Value("${storage.root}")
+    private String storageRoot;
+
+    private Path storageRootPath;
+
+    private Path temporaryDir;
+
+    @Value("${storage.file.max-size.mb}")
+    private long fileMaxSizeMb;
+
+    private final FileDataRepo fileDataRepo;
+
+    public FileService(FileDataRepo fileDataRepo, MongoTemplate mongoTemplate) {
+        this.fileDataRepo = fileDataRepo;
+    }
+
+    @PostConstruct
+    public void init(){
+        storageRootPath = Paths.get(storageRoot);
+        if (Files.notExists(storageRootPath)){
+            try {
+                Files.createDirectories(storageRootPath);
+            } catch (IOException e) {
+                log.error(String.format("It is impossible to create storage root directory : %s", storageRoot), e);
+                System.exit(-1);
+            }
+        }
+    }
+
+    public FileData uploadFile(UploadFileDTO uploadFileDTO) throws FileIsToLargeException{
+        try {
+            URL url = new URL(uploadFileDTO.getFileUrl());
+            String filename = UUID.randomUUID().toString() + "." + FilenameUtils.getExtension(url.toString());
+            temporaryDir = Files.createTempDirectory(storageRootPath, "temp");
+            Path path = storageRootPath.resolve(filename);
+            File tempFile = uploadFileFromUrl(url);
+
+            FileData fileData = FileData.builder().originalFilename(uploadFileDTO.getFileName())
+                    .filename(filename)
+                    .size(Files.size(tempFile.toPath()))
+                    .type(Files.probeContentType(path))
+                    .directory("/")
+                    .build();
+
+            fileDataRepo.insert(fileData);
+            Files.move(tempFile.toPath(), path);
+            Files.deleteIfExists(tempFile.toPath());
+            Files.deleteIfExists(temporaryDir);
+            return fileData;
+        } catch (IOException e) {
+            log.error("An error occurred during uploading file");
+            return null;
+        }
+    }
+
+    public List<FileData> getFileListFromDirectory(String directory){
+        String regex = "^/" + directory;
+        return fileDataRepo.findFileDataByRegexpDirectory(regex);
+    }
+
+    public void removeFile(String id) throws FileNotFoundException {
+        Optional<FileData> optionalFileData = fileDataRepo.findById(id);
```

Comment:
I suggest small improvment - new method:
```
FileData getFileById(String id) {
   Optional<FileData> optionalFileData = fileDataRepo.findById(id);
   if (optionalFileData.isPresent()) {
      return optionalFileData.get();
  } else {
      thow new FileNotFoundException(...);
  }
}
```
So you can use it from:
* removeFile(String id);
* moveFile(String id, String directory);
* copyFile(String id, String directory);
* getFileContent(String id, HttpServletResponse response);